cdap-app-fabric-tests/src/test/java/io/cdap/cdap/internal/app/runtime/batch/DynamicPartitionerWithAvroTest.java

```
@@ -172,6 +172,10 @@ public void apply() throws IOException {
     // run the partition writer m/r with this output partition time
     Map<String, String> arguments = new HashMap<>();
     arguments.put(OUTPUT_PARTITION_KEY, Long.toString(now));
+    // The test case and the output committer was written in hadoop 2, which followed older file output committer
+    // algorithm ( version 1 ). After upgrading to Hadoop 3, the default algorithm is version 2. So setting it manually
+    // to previous algorithm version to comply with with test.
```

Why are we fixing it for test only? Test just highlights the problem, but the issue of different file names would mean prod code won't work either.

---

Also please do a manual test of multioutput GCS plugin. This may have impact on it

---

I tested ` multioutput GCS` with avro n parquet and compared with an 6.10 instance. The output behavior is same.
it was `BQ` --> `multi gcs` 

As per [Hadoop Doc](https://hadoop.apache.org/docs/r3.1.1/hadoop-mapreduce-client/hadoop-mapreduce-client-core/mapred-default.xml) , the logical output is the same. The way of handling temporary files and moving them around has changed. 

Also AFAIK , we don't ship Hadoop / spark jars to dataproc. So, this change of algorithm is already there in dataproc 2.0 + .. 

What do you suggest ?

---

Please address this comment

---

As discussed offline, we need to fail fast if we encounter version 2. 

Added the check in `DynamicPartitioningOutputCommitter.java`